keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/MultivariateGaussian.java

```diff
@@ -0,0 +1,95 @@
+package io.improbable.keanu.vertices.dbl.probabilistic;
+
+import io.improbable.keanu.distributions.tensors.continuous.TensorMultivariateGaussian;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.dbl.Nd4jDoubleTensor;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.ConstantDoubleVertex;
+import org.apache.commons.math3.linear.MatrixUtils;
+import org.apache.commons.math3.linear.RealMatrix;
+
+import java.util.Map;
+
+public class MultivariateGaussian extends ProbabilisticDouble {
+
+    private final DoubleVertex mu;
+    private final DoubleVertex covariance;
+    private final GaussianVertex variates;
+
+    /**
+     * Multivariate gaussian distribution. The shape is driven from mu, which must be a vector.
+     * The shape of the covariance (matrix) must be a square that is the same height as mu.
+     *
+     * @param shape      the desired shape of the vertex
+     * @param mu         the mu of the Multivariate Gaussian
+     * @param covariance the covariance matrix of the Multivariate Gaussian
+     */
+    public MultivariateGaussian(int[] shape, DoubleVertex mu, DoubleVertex covariance) {
+
+        checkValidMultivariateShape(mu.getShape(), covariance.getShape());
+
+        this.mu = mu;
+        this.covariance = covariance;
+        this.variates = new GaussianVertex(mu.getShape(), 0, 1);
+        setParents(mu, covariance);
+        setValue(DoubleTensor.placeHolder(shape));
+    }
+
+    /**
+     * Matches a mu and covariance of some shape to a Multivariate Gaussian
+     *
+     * @param mu         the mu of the Multivariate Gaussian
+     * @param covariance the covariance matrix of the Multivariate Gaussian
+     */
+    public MultivariateGaussian(DoubleVertex mu, DoubleVertex covariance) {
+        this(checkValidMultivariateShape(mu.getShape(), covariance.getShape()), mu, covariance);
+    }
+
+
+    /**
+     * Matches a mu to a Multivariate Gaussian. The covariance value provided here
+     * is used to create a covariance tensor by multiplying the scalar value against
+     * an identity matrix of the appropriate size.
+     *
+     * @param mu         the mu of the Multivariate Gaussian
+     * @param covariance the scale of the identity matrix
+     */
+    public MultivariateGaussian(DoubleVertex mu, double covariance) {
+        this(mu, new ConstantDoubleVertex(Nd4jDoubleTensor.eye(mu.getShape()[0])).times(covariance));
```

Comment:
Go for DoubleTensor.eye so that in the case where `mu.getShape()[0] == 1` you get a simple scalar mu instead of a nd4j scalar.

Reply:
Nice, changed